stakingModel/test/SimpleDAIStaking.e2e.test.js

```diff
@@ -0,0 +1,72 @@
+const SimpleDAIStaking = artifacts.require("SimpleDAIStaking");
+const GoodDollar = artifacts.require("GoodDollar");
+const DAIMock = artifacts.require("DAIMock");
+const cDAIMock = artifacts.require("cDAIMock");
+
+const fse = require("fs-extra");
+
+const BN = web3.utils.BN;
+export const BLOCK_INTERVAL = 2;
+export const NULL_ADDRESS = "0x0000000000000000000000000000000000000000";
+export const DAI_ADDRESS  = "0x4F96Fe3b7A6Cf9725f59d353F723c1bDb64CA6Aa";
+export const cDAI_ADDRESS = "0xe7bc397dbd069fc7d0109c0636d06888bb50668c";
+
+contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
+  let dai;
+  let cDAI;
+  let simpleStaking;
+
+  before(async () => {
+    dai = await DAIMock.at(DAI_ADDRESS);
+    cDAI = await cDAIMock.at(cDAI_ADDRESS);
+    simpleStaking = await SimpleDAIStaking.new(
+      DAI_ADDRESS,
+      cDAI_ADDRESS,
+      NULL_ADDRESS,
+      founder,
+      BLOCK_INTERVAL
+    );
+  });
+
+  it("should be able to stake dai, withdraw gains if exists, and withdraw the staked tokens", async () => {
+    await dai.approve(simpleStaking.address, web3.utils.toWei("500", "gwei"), {
+      from: staker
+    });
+    let balanceBefore = (await simpleStaking.stakers(staker)).stakedDAI;
+    let totalStakedBefore = await simpleStaking.totalStaked();
+    await simpleStaking
+      .stakeDAI(web3.utils.toWei("500", "gwei"), {
+        from: staker
+      })
+      .catch(console.log);
+    let balanceAfter = (await simpleStaking.stakers(staker)).stakedDAI;
+    expect((balanceAfter - balanceBefore).toString()).to.be.equal(
+      web3.utils.toWei("500", "gwei")
+    );
+    let totalStakedAfter = await simpleStaking.totalStaked();
+    expect((totalStakedAfter - totalStakedBefore).toString()).to.be.equal(
+      web3.utils.toWei("500", "gwei")
+    );
+    const gains = await simpleStaking.currentUBIInterest();
+    const cdaiGains = gains["0"];
+    const fundBalanceBefore = await cDAI.balanceOf(founder);
+    await simpleStaking.collectUBIInterest(founder);
```

Comment:
are we gonna have any gains? maybe mint very large amount of DAI and wait a few blocks?
is it possible to test this on a test network in a unit test? how can we actually test cDAI generating revenues

Reply:
?